ucp/_libs/ucp_tiny.pyx

```diff
@@ -0,0 +1,329 @@
+# Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+# See file LICENSE for terms.
+# cython: language_level=3
+
+import asyncio
+import uuid
+from ucp_tiny_dep cimport *
+
+
+def assert_error(exp, msg=""):
+    if not exp:
+        raise AssertionError(msg)
+
+
+cdef struct _listener_callback_args:
+    ucp_worker_h ucp_worker
+    void *py_func
```

Comment:
Instead of making this a `void *`, would it make sense to keep this as an `object` so Cython can handle the reference counting for us and only cast this to `void *` when C needs one for some reason? Suspect this will save us from running into subtle reference counting bugs.

Reply:
Turns out one cannot have an `object` in a `struct`. Still wonder if there is a way for us to have the `struct` and avoid explicit reference counting.

Reply:
Yeah, I would love to know if there is a technique we can use.

Reply:
We should be able to get a PyObject* and put that in the struct, no? That would require interfacing to the Python C-API though which can get hairy.